README.md

```diff
@@ -59,35 +59,30 @@ This will start the development server under
 
 ```bash
 git clone git@github.com:oceanprotocol/barge.git
 cd barge
 
-# startup with local Ganache node
-./start_ocean.sh
+# startup with local Ganache and Graph nodes
+./start_ocean.sh --with-thegraph
 ```
 
-Barge will deploy contracts to the local Ganache node which will take some time. At the end the compiled artifacts need to be copied over to this project into `node_modules/@oceanprotocol/contracts/artifacts`. This script will do that for you:
+Barge will deploy contracts to the local Ganache node which will take some time. At the end the compiled artifacts need to imported over to this project as environment variables. The `set-barge-env` script will do that for you and set the env variables to use this local connection in `.env` in the app.
 
-```bash
-./scripts/copy-contracts.sh
-```
+If you are using `macOS` operating system you should also make same changes to the provider url since the default barge ip can not be accessed due to some network constraints on `macOs`. So we should be using the `127.0.0.1:8030` (if you have changed the provider port please use that here as well) for each direct call from the market to provider, but we should keep the internal barge url `http://172.15.0.4:8030/` (this is the default ip:port for provider in barge, if changed please use the according url). So on inside `src/@utils/provider.ts` if on `macOS` you can hardcode this env variable `NEXT_PUBLIC_PROVIDER_URL` or set
+`127.0.0.1:8030` as `providerUrl` on all the methods that call `ProviderInstance` methods. (eg: `getEncryptedFiles`, `getFileDidInfo`, `downloadFile` etc). You should use the same provider url for `src/@utils/nft.ts` inside `setNFTMetadataAndTokenURI` and `setNftMetadata` and `src/components/Publish/index.tsx` inisde `encrypt` method (if you set the env variable there's no need to do this). You also need to use local ip's for the subgraph (`127.0.0.1` instead of `172.15.0.15`) and the metadatacache (`127.0.0.1` instead of `172.15.0.5`).
 
-Finally, set environment variables to use this local connection in `.env` in the app:
+If you are using the `subgraph` inside `barge` as well and there are not approved tokens inside your local subgraph you should either insert them or harcode the ocean token address from this env variable `process.env.NEXT_PUBLIC_OCEAN_TOKEN_ADDRESS` inside `src/components/Publish/_utils.ts` as feeToken and also set the `baseTokenDecimals` to `18`.
 
 ```bash
-# modify env variables
-cp .env.example .env
-
+npm run set-barge-env
 npm start
 ```
 
 To use the app together with MetaMask, importing one of the accounts auto-generated by the Ganache container is the easiest way to have test ETH available. All of them have 100 ETH by default. Upon start, the `ocean_ganache_1` container will print out the private keys of multiple accounts in its logs. Pick one of them and import into MetaMask.
```

Comment:
The accounts used in barge are always the same so lets paste the private keys here so that users don't have to look through the console to find the private keys

Reply:
To avoid duplication think we can just link the barge readme section where they are written down.